src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -866,31 +866,16 @@ public virtual bool HasUnsupportedMetadata
 
         internal bool? GetNonNullTypesFromSyntax()
         {
-            bool? result = null;
-            string id = "CS" + ((int)ErrorCode.WRN_PragmaNonNullTypes).ToString("0000");
-
             foreach (Location location in Locations)
             {
                 SyntaxTree tree = location.SourceTree;
-
                 if (tree != null)
                 {
-                    ReportDiagnostic? state = ((CSharpSyntaxTree)tree).GetPragmaDirectiveSpecificWarningState(id, location.SourceSpan.Start);
-                    if (state.HasValue)
-                    {
-                        if (state.GetValueOrDefault() == ReportDiagnostic.Suppress)
-                        {
-                            result = false;
-                        }
-                        else
-                        {
-                            return true;
-                        }
-                    }
+                    return ((CSharpSyntaxTree)tree).GetNonNullDirectiveState(location.SourceSpan.Start);
```

Comment:
>return ((CSharpSyntaxTree)tree).GetNonNullDirectiveState(location.SourceSpan.Start); [](start = 20, length = 84)

It looks like semantics of merging across different partial type declaration is not preserved. The lack of tests that catch this is probably my fault.